pkg/controllers/nodeclaim/disruption/controller.go

```diff
@@ -58,47 +57,39 @@ type Controller struct {
 }
 
 // NewController constructs a machine disruption controller
-func NewController(clk clock.Clock, kubeClient client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider) corecontroller.Controller {
-	return corecontroller.Typed[*v1alpha5.Machine](kubeClient, &Controller{
+func NewController(clk clock.Clock, kubeClient client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider) *Controller {
+	return &Controller{
 		kubeClient: kubeClient,
 		drift:      &Drift{cloudProvider: cloudProvider},
 		expiration: &Expiration{kubeClient: kubeClient, clock: clk},
 		emptiness:  &Emptiness{kubeClient: kubeClient, cluster: cluster, clock: clk},
-	})
-}
-
-func (c *Controller) Name() string {
-	return "machine.disruption"
+	}
 }
 
 // Reconcile executes a control loop for the resource
-func (c *Controller) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (reconcile.Result, error) {
-	stored := machine.DeepCopy()
-	if _, ok := machine.Labels[v1alpha5.ProvisionerNameLabelKey]; !ok {
-		return reconcile.Result{}, nil
+func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeClaim) (reconcile.Result, error) {
+	stored := nodeClaim.DeepCopy()
+	nodePool, err := nodeclaimutil.Owner(ctx, c.kubeClient, nodeClaim)
```

Comment:
Does it make sense for us to isolate the compatibility logic into a package named `compat`, to help us remove things in the future.

Reply:
This is a good point. I can make a separate package in a separate PR but changing that naming is probably beyond the scope of this PR